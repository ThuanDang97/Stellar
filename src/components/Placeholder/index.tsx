import React, { memo } from 'react'

// Components
import { ViewStyled, RowStyled } from '@components/styled-components/ViewStyled'
import PlaceholderItem from '@components/PlaceholderItem'
import { useTheme } from 'styled-components'

// Props type
import { PlaceholderProps } from '@self-types/components/Placeholder.props'

const Placeholder = ({ isLinkBill = true }: PlaceholderProps) => {
  const theme = useTheme()

  return (
    <RowStyled>
      <PlaceholderItem
        type="rect"
        width={theme.metrics.width.base}
        height={theme.metrics.height.base}
      />
      <ViewStyled
        mLeft={theme.metrics.dimensions.sm}
        mRight={theme.metrics.dimensions.xxl}
      >
        <PlaceholderItem width={theme.metrics.width.md} />
        {isLinkBill && (
          <PlaceholderItem
            width={theme.metrics.width.lg}
            height={theme.metrics.height.tiny}
            mTop={theme.metrics.dimensions.xs}
          />
        )}
      </ViewStyled>
      <ViewStyled mRight={theme.metrics.dimensions.xxl}>
        <PlaceholderItem width={theme.metrics.width.lg} />
      </ViewStyled>
      <ViewStyled mRight={theme.metrics.dimensions.xl}>
        <PlaceholderItem />
      </ViewStyled>
      <PlaceholderItem width={13} />
    </RowStyled>
  )
}

export default memo(Placeholder)
