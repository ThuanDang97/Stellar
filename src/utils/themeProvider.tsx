import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
