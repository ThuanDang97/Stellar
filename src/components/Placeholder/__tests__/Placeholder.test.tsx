import '@testing-library/jest-dom'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// component
import Placeholder from '../index'

describe('Placeholder Component', () => {
  it('renders placeholder component with link bill', () => {
    const { container } = renderWithTheme(<Placeholder isLinkBill />)
    expect(container).toMatchSnapshot()
  })

  it('renders placeholder component with goal', () => {
    const { container } = renderWithTheme(<Placeholder isLinkBill={false} />)
    expect(container).toMatchSnapshot()
  })
})
