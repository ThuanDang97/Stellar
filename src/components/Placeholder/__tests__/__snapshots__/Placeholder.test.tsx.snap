// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Component renders placeholder component with goal 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__RowStyled-sc-rqe5sa-1 fTsvHV eRHuZf"
  >
    <div
      class="rect-shape"
      style="background-color: rgb(230, 230, 230); width: 2.5rem; height: 2.5rem; margin-right: 10px; margin-top: 0rem;"
    />
    <div
      class="ViewStyled-sc-rqe5sa-0 fNNXne"
    >
      <div
        class="text-row"
        style="width: 4.375rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 bwSFDl"
    >
      <div
        class="text-row"
        style="width: 8.75rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 hyFUvf"
    >
      <div
        class="text-row"
        style="width: 3.125rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
    </div>
    <div
      class="text-row"
      style="width: 0.8125rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
    />
  </div>
</div>
`;

exports[`Placeholder Component renders placeholder component with link bill 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__RowStyled-sc-rqe5sa-1 fTsvHV eRHuZf"
  >
    <div
      class="rect-shape"
      style="background-color: rgb(230, 230, 230); width: 2.5rem; height: 2.5rem; margin-right: 10px; margin-top: 0rem;"
    />
    <div
      class="ViewStyled-sc-rqe5sa-0 fNNXne"
    >
      <div
        class="text-row"
        style="width: 4.375rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
      <div
        class="text-row"
        style="width: 8.75rem; height: 0.375rem; background-color: rgb(230, 230, 230); margin-top: 0.3125rem;"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 bwSFDl"
    >
      <div
        class="text-row"
        style="width: 8.75rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 hyFUvf"
    >
      <div
        class="text-row"
        style="width: 3.125rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
      />
    </div>
    <div
      class="text-row"
      style="width: 0.8125rem; height: 0.8125rem; background-color: rgb(230, 230, 230); margin-top: 0rem;"
    />
  </div>
</div>
`;
